File: p4/p4.js
// Find the largest palindome number (9009) that is the product of 2 3-diget numbers.

function is_pal(input){
  var string = input.toString();
  var len = string.length;

  for (var i = 0; i < len; i++) {
    if (string[i] != string[len - (1 + i)]) {
      return false
    }
  }
  return true
}

var output = 0;

for (var x = 999; x > 99; x--) {
  for (var y = 999; y > 99; y--) {
    var current = x * y;
    if (current > output && is_pal(current)) {
      output = current
    }
  }
}

console.log(output)
// 906609
// 913 * 993

File: p4/p4.clj
; look for the largest palindome number (9009) that
; is the product of 2 3-digit numbers.

(use '[clojure.string :only (join split)])

(defn is_pal [input]
  (= (str input) (join (reverse (str input)))))

(defn loop_check [input]
  (loop [x 999]
    (if (< 0 x)
      (if (is_pal (* x input))
        (* x input)
        (recur (dec x))))))

(loop [x 999 best 0]
  (if (< 0 x)
    (let [current (or (loop_check x) 0)]
      (recur (dec x) (max best current)))
    (print best)))

; 906609
; from 913 993
